homeassistant/components/imeon_inverter/sensor.py

```diff
@@ -0,0 +1,449 @@
+        key="monitoring_solar_production",
+        translation_key="monitoring_solar_production",
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    # Monitoring (instant minute data)
+    SensorEntityDescription(
+        key="monitoring_minute_building_consumption",
+        translation_key="monitoring_minute_building_consumption",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="monitoring_minute_grid_consumption",
+        translation_key="monitoring_minute_grid_consumption",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="monitoring_minute_grid_injection",
+        translation_key="monitoring_minute_grid_injection",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="monitoring_minute_grid_power_flow",
+        translation_key="monitoring_minute_grid_power_flow",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="monitoring_minute_solar_production",
+        translation_key="monitoring_minute_solar_production",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: InverterConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Create each sensor for a given config entry."""
+
+    # Get Inverter from UUID
+    coordinator = entry.runtime_data
+
+    # Init sensor entities
+    async_add_entities(
+        InverterSensor(coordinator, entry, description)
+        for description in ENTITY_DESCRIPTIONS
+    )
+
+
+class InverterSensor(CoordinatorEntity[InverterCoordinator], SensorEntity):
+    """A sensor that returns numerical values with units."""
+
+    entity_description: SensorEntityDescription
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: InverterCoordinator,
+        entry: InverterConfigEntry,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Pass coordinator to CoordinatorEntity."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self.data_key = description.key
+        self._attr_translation_key = description.key
+        self._attr_unique_id = f"{entry.entry_id}_{self.data_key}"
```

Comment:
```suggestion
        assert entry.unique_id
        self._attr_unique_id = f"{entry.unique_id}_{self.data_key}"
```

you are setting a unique id so let's use that. You'll need to update your snapshots, but that'll also fix the issue that the entity unique id is currently changing between tests and breaks tests